### PR TITLE
FMWK-255 Add qualifier builders

### PR DESCRIPTION
**Pull request by agrgr**
Add builder for parent qualifier containing other qualifiers, add builder for id qualifier (to have a dedicated map key), return immutable map in BaseQualifierBuilder, add some refactoring and cleanup.

### CODE REVIEW

File: src/main/java/org/springframework/data/aerospike/query/Qualifier.java [hunk 1]
```diff
@@ -381,28 +431,39 @@ protected void validate() {
...
     }
+
+    /**
+     * Create a parent qualifier that contains the given qualifiers connected using logical AND/OR
+     *
+     * @param logicalOp {@link FilterOperation#AND} or {@link FilterOperation#OR}
+     * @param qualifiers 2 or more qualifiers to be connected
+     * @return Parent qualifier
+     */
+    public static Qualifier forMultipleQualifiers(FilterOperation logicalOp, Qualifier... qualifiers) {
```
Comment (roimenashe) [inline src/main/java/org/springframework/data/aerospike/query/Qualifier.java:464]: why the "for" prefix? why not just `multipleQualifiers`?

Comment (reugn) [reply src/main/java/org/springframework/data/aerospike/query/Qualifier.java:464]:
```java
public static Qualifier or(Qualifier qualifier, Qualifier... qualifiers)
public static Qualifier and(Qualifier qualifier, Qualifier... qualifiers)
```

File: src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java [hunk 2]
```diff
@@ -146,7 +141,14 @@ public AerospikeCriteria getCriteria(Part part, AerospikePersistentProperty prop
         String dotPath = null;
         Object value3 = null;
 
-        if (property.isCollectionLike()) {
+        if (property.isIdProperty()) {
+            if (value1 instanceof Collection<?>) {
+                return new AerospikeCriteria(forIds(((Collection<?>) value1).toArray(String[]::new)));
+            } else {
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java:147]: remove else

File: src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java [hunk 3]
```diff
@@ -379,18 +373,13 @@ protected AerospikeCriteria and(Part part, AerospikeCriteria base, Iterator<Obje
...
     }
 
     @Override
     protected AerospikeCriteria or(AerospikeCriteria base, AerospikeCriteria criteria) {
-        return new AerospikeCriteria(Qualifier.builder()
-            .setFilterOperation(FilterOperation.OR)
-            .setQualifiers(base, criteria)
-        );
+        return new AerospikeCriteria(forMultipleQualifiers(OR, base, criteria));
+
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/repository/query/AerospikeQueryCreator.java:382]: remove line

File: src/main/java/org/springframework/data/aerospike/query/Qualifier.java [hunk 4]
```diff
@@ -108,7 +117,15 @@ public boolean hasQualifiers() {
     }
 
     public boolean hasId() {
-        return internalMap.get(FIELD) != null && internalMap.get(FIELD).equals(ID_VALUE);
+        return internalMap.get(SINGLE_ID_FIELD) != null ||  internalMap.get(MULTIPLE_IDS_FIELD) != null ;
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/query/Qualifier.java:120]:
```suggestion
        return internalMap.get(SINGLE_ID_FIELD) != null || internalMap.get(MULTIPLE_IDS_FIELD) != null;
```

File: src/main/java/org/springframework/data/aerospike/query/Qualifier.java [hunk 5]
```diff
@@ -381,28 +431,50 @@ protected void validate() {
     }
 
     /**
-     * Create qualifier "ID is equal to the given string"
+     * Create a qualifier "ID is equal to the given string"
      *
      * @param id String value
      * @return Single id qualifier
      */
     public static Qualifier forId(String id) {
```
Comment (roimenashe) [inline src/main/java/org/springframework/data/aerospike/query/Qualifier.java:439]: Same here I think, not sure about the "for" prefix maybe `IdEquals` and `IdIn()`? lets talk about it tomorrow

**Review by reugn — APPROVED** [review]

**Pull request merged**